### PR TITLE
feat: implement Device SDK v2 common API

### PR DESCRIPTION
**Pull request by hahattan**
for consistency across services, all implementation are all reference to
https://github.com/edgexfoundry/app-functions-sdk-go/tree/master/internal/v2

this PR need to be merged before other v2 API PRs so that the controller can be reused.

Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-CommitMessages
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### CODE REVIEW

File: internal/v2/controller/http/common.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// It returns a response as specified by the V2 API swagger in openapi/v2
+func (c *V2HttpController) Ping(writer http.ResponseWriter, request *http.Request, _ *di.Container) {
+	response := common.NewPingResponse()
+	c.sendResponse(writer, request, contractsV2.ApiPingRoute, response, http.StatusOK)
+}
+
+// Version handles the request to /version endpoint. Is used to request the service's versions
+// It returns a response as specified by the V2 API swagger in openapi/v2
+func (c *V2HttpController) Version(writer http.ResponseWriter, request *http.Request, _ *di.Container) {
+	response := common.NewVersionResponse(sdkCommon.V2ServiceVersion)
```
Comment (lenny-intel) [inline internal/v2/controller/http/common.go:24]: Needs to use common.NewVersionSdkResponse() and include the SDKversion

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/common/consts.go [hunk 2]
```diff
@@ -40,13 +38,27 @@ const (
...
 	IdVar        string = "id"
 	NameVar      string = "name"
 	CommandVar   string = "command"
 	GetCmdMethod string = "get"
 	SetCmdMethod string = "set"
 
-	CorrelationHeader = clients.CorrelationHeader
-	URLRawQuery       = "urlRawQuery"
-	SDKReservedPrefix = "ds-"
+	V2CorrelationHeaderKey = "X-Correlation-ID"
```
[the remark this answers is not in the document]
Comment (hahattan) [reply internal/common/consts.go:57]: `clients.CorrelationHeader` is `correlation-id` currently
https://github.com/edgexfoundry/go-mod-core-contracts/blob/a48a3cfc95ac7623bae8df3d1eb5020ec58becf8/clients/constants.go#L24, will it be updated in the future ?

[the remark this answers is not in the document]
Comment (lenny-intel) [reply internal/common/consts.go:57]: Dang, I thought that one was the correct one. I see the the App SDK, we have a similar constant. Yes, we need to get this cleaned up so there is one correct constant, but will probably have to wait till Ireland 2.0 to fix this. So for now your constant is fine.

File: internal/common/consts.go [hunk 3]
```diff
@@ -40,13 +38,27 @@ const (
...
+	CorrelationHeader      = clients.CorrelationHeader
+	URLRawQuery            = "urlRawQuery"
+	SDKReservedPrefix      = "ds-"
 )
+
+// ServiceVersion indicates the version of the device service itself, not the SDK - will be overwritten by build
+var V2ServiceVersion string = "0.0.0"
```
[the remark this answers is not in the document]
Comment (hahattan) [reply internal/common/consts.go:64]: updated.

File: internal/v2/controller/http/common.go (hunk 1 above)
Comment (hahattan) [reply internal/v2/controller/http/common.go:24]: updated.

Comment (hahattan) [reply internal/v2/controller/http/common.go]: noticed that although in [this discussion](https://github.com/edgexfoundry/device-sdk-go/pull/549#discussion_r460215570) we decided to leave it fully until v2, but seems passing `SDKVersion` during build time won't break any compatibility so I'll just make the change now.

Comment (lenny-intel) [reply internal/v2/controller/http/common.go]: Correct.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**